.github/workflows/generate-from-configuration.yaml

```diff
@@ -0,0 +1,113 @@
+        # be referenced by the container and its children
+        if [[ $(docker volume inspect repo-google-cloud-java) != '[]' ]]; then
+          docker volume rm repo-google-cloud-java
+        fi
+        docker volume create --name "repo-google-cloud-java" --opt "type=none" --opt "device=$(pwd)" --opt "o=bind"
+    - name: update googleapis commit to latest
+      id: update-commit
+      shell: bash
+      run: |
+        mkdir tmp-googleapis
+        # use partial clone because only commit history is needed.
+        git clone --filter=blob:none https://github.com/googleapis/googleapis.git tmp-googleapis
+        pushd tmp-googleapis
+        git pull
+        latest_commit=$(git rev-parse HEAD)
+        popd
+        rm -rf tmp-googleapis
+        sed -i -e "s/^googleapis_commitish.*$/googleapis_commitish: ${latest_commit}/" generation_config.yaml
+    - name: generate from configuration
+      shell: bash
+      run: |
+        docker run --rm \
+          ${repo_volumes} \
+          -v /tmp:/tmp \
+          -v /var/run/docker.sock:/var/run/docker.sock \
+          -e "RUNNING_IN_DOCKER=true" \
+          -e "REPO_BINDING_VOLUMES=${repo_volumes}" \
+          gcr.io/cloud-devrel-public-resources/java-library-generation:"${library_generation_image_tag}" \
+          python /src/generate_repo.py generate --generation-config-yaml=/workspace/google-cloud-java/generation_config.yaml --repository-path=/workspace/google-cloud-java
+    - name: generate pull request description
+      id: generate-description
+      shell: bash
+      run: |
+        # the pr description (body) will be available in pr-description.txt after
+        # running this command.
+        set -x
+        docker run --rm \
+          ${repo_volumes} \
+          -v /tmp:/tmp \
+          -v /var/run/docker.sock:/var/run/docker.sock \
+          -e "RUNNING_IN_DOCKER=true" \
+          -e "REPO_BINDING_VOLUMES=${repo_volumes}" \
+          gcr.io/cloud-devrel-public-resources/java-library-generation:"${library_generation_image_tag}" \
+          python /src/generate_pr_description.py generate --generation-config-yaml=/workspace/google-cloud-java/generation_config.yaml --baseline-commit="${baseline_commit}"
+      env:
+        baseline_commit: ${{ env.baseline_commit }}
+    - name: create or update the pull request
+      shell: bash
+      run: |
+        title="chore: generate libraries at $(date)"
+        git add java-* pom.xml gapic-libraries-bom/pom.xml versions.txt generation_config.yaml
+        # use --allow-empty because (rarely) there's no change.
+        git commit --allow-empty -m "${title}"
+        if [ -z "${pr_num}" ]; then
+          git remote add monorepo https://cloud-java-bot:${GH_TOKEN}@github.com/${{ github.repository }}.git
+          git fetch -q --unshallow monorepo
+          git push -f monorepo "${branch_name}"
+          set -x
+          gh pr create --title "${title}" --head "${branch_name}" --body "$(cat pr_description.txt)"
```

Comment:
I guess this creates a PR against main branch by default? Can we make the target branch a parameter as well?

Reply:
Done.